js/userDatabase.js

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+(function(exports) {
+  const DATABASE_VERSION = 1;
+  const DATABASE_NAME = "NewTabData";
+
+  const OBJECT_STORE_PREFS = "prefs";
+  const PINNED_LINKS_PREF = "pinnedLinks";
+
+  const gUserDatabase = {
+    _database: null,
+
+    init(mockDB) {
+      return new Promise((resolve, reject) => {
+        var db = mockDB || window.indexedDB;
+        var request = db.open(DATABASE_NAME, DATABASE_VERSION);
```

Comment:
this is only ever called once, so we don't need the constants, just use strings here.